core/src/main/java/org/opensearch/sql/executor/execution/PaginatedQueryService.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.executor.execution;
+
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.analysis.AnalysisContext;
+import org.opensearch.sql.analysis.Analyzer;
+import org.opensearch.sql.ast.tree.Paginate;
+import org.opensearch.sql.ast.tree.UnresolvedPlan;
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.executor.ExecutionContext;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.planner.Planner;
+import org.opensearch.sql.planner.logical.LogicalPlan;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+
+@RequiredArgsConstructor
+public class PaginatedQueryService {
```

Comment:
why not reuse QueryService?

Reply:
The difference between `PaginatedQueryService` and `QueryService` is in different Planners. You can see it [here](https://github.com/Bit-Quill/opensearch-project-sql/blob/ca76e1b9248fc87b752bd4a4d12cd0fea566ac7e/plugin/src/main/java/org/opensearch/sql/plugin/config/OpenSearchPluginModule.java#L109-L113).
Options to do:
1. Do nothing, keep as is.
2. Add second `Planner` to `QueryService` and use it.
3. Create `QueryService` interface and 2 implementations with different planners.

Reply:
I can do everything!
Please, clarify - do you want to see `QueryService` with 2 planners?

Reply:
Fixed in 75b814078.